src/utils/runtime/src/logger.rs

```diff
@@ -247,53 +247,73 @@ pub fn init_risingwave_logger(settings: LoggerSettings) {
                 e.as_report(),
             )
         });
-        let file = std::fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(query_log_path.join("query.log"))
-            .unwrap_or_else(|e| {
-                panic!(
-                    "failed to create '{}/query.log': {}",
-                    query_log_path.display(),
-                    e.as_report(),
-                )
-            });
-        let layer = tracing_subscriber::fmt::layer()
-            .with_ansi(false)
-            .with_level(false)
-            .with_file(false)
-            .with_target(false)
-            .with_timer(default_timer.clone())
-            .with_thread_names(true)
-            .with_thread_ids(true)
-            .with_writer(std::sync::Mutex::new(file))
-            .with_filter(filter::Targets::new().with_target(PGWIRE_QUERY_LOG, Level::TRACE));
-        layers.push(layer.boxed());
 
-        let file = std::fs::OpenOptions::new()
-            .create(true)
-            .write(true)
-            .truncate(true)
-            .open(query_log_path.join("slow_query.log"))
-            .unwrap_or_else(|e| {
-                panic!(
-                    "failed to create '{}/slow_query.log': {}",
-                    query_log_path.display(),
-                    e.as_report(),
-                )
-            });
-        let layer = tracing_subscriber::fmt::layer()
-            .with_ansi(false)
-            .with_level(false)
-            .with_file(false)
-            .with_target(false)
-            .with_timer(default_timer)
-            .with_thread_names(true)
-            .with_thread_ids(true)
-            .with_writer(std::sync::Mutex::new(file))
-            .with_filter(filter::Targets::new().with_target(SLOW_QUERY_LOG, Level::TRACE));
-        layers.push(layer.boxed());
+        /// Newtype wrapper for `DefaultFields`.
+        ///
+        /// `fmt::Layer` will share the same `FormattedFields` extension for spans across
+        /// different layers, as long as the type of `N: FormatFields` is the same. This
+        /// will cause several problems:
+        ///
+        /// - `with_ansi(false)` does not take effect and it will follow the settings of
+        ///   the primary fmt layer installed above.
+        /// - `Span::record` will update the same `FormattedFields` multiple times,
+        ///   leading to duplicated fields.
+        ///
+        /// As a workaround, we use a newtype wrapper here to get a different type id.
+        /// The const generic parameter `SLOW` is further used to distinguish between the
+        /// query log and the slow query log.
```

Comment:
Specifically, what part of the following log is controlled by this?
```
2023-12-12T16:43:36.017757+08:00 risingwave-main ThreadId(10) handle_query{mode="simple query" session_id=0 sql=select fu;}: status="err" time=68.853375ms

```

Reply:
```diff
- handle_query
+ <some control character for ANSI color>handle_query
- {mode="simple query" session_id=0 sql=select fu;}:
+ {mode="simple query" session_id=0 sql=select fu; sql=select fu; sql=select fu;}:
 status="err" time=68.853375ms
```

Reply:
Why it's colorful: https://docs.rs/tracing-subscriber/0.3.18/src/tracing_subscriber/fmt/fmt_layer.rs.html#815-823

Why `sql` is recorded multiple times: https://docs.rs/tracing-subscriber/0.3.18/src/tracing_subscriber/fmt/fmt_layer.rs.html#851-854

Notice the type parameter `N` here.